api/v1/schedule_types.go

```diff
@@ -83,6 +83,13 @@ type ScheduleStatus struct {
 	// They are an extension mechanism which allows tools and other controllers to collect summary information about
 	// resources without needing to understand resource-specific status details.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// EffectiveSchedules contains a list of schedules generated from randomizing schedules.
+	EffectiveSchedules []EffectiveSchedule `json:"effectiveSchedules,omitempty"`
```

Comment:
I know that for specs maps are an antipattern for K8s.

But I feel here for the status it would simplify quite a few things?

e.g. map[JobType]EffectiveSchedule?

Reply:
It's not that bad, there's just about 30 LoC that deal with setting/removing effective schedules, not worth it IMO. I wouldn't recommend using an antipattern just to get rid of 30 lines.

Reply:
Fine by me.